tests/test_web_urldispatcher.py

```diff
@@ -458,3 +459,47 @@ async def test_static_absolute_url(aiohttp_client: Any, tmp_path: Any) -> None:
     client = await aiohttp_client(app)
     resp = await client.get("/static/" + str(file_path.resolve()))
     assert resp.status == 403
+
+
+@pytest.mark.xfail(
+    raises=AssertionError,
+    reason="Regression in v3.7: https://github.com/aio-libs/aiohttp/issues/5621",
+)
+async def test_decoded_match_regex(aiohttp_client) -> None:
+    app = web.Application()
+
+    async def handler(_):
+        return web.Response()
+
+    app.router.add_get(
+        "/{user_ids:(([0-9]+)|(u_[0-9a-f]{16}))(,(([0-9]+)|(u_[0-9a-f]{16})))*}/hello",
+        handler,
+    )
+    client = await aiohttp_client(app)
+
+    # '/467,802,24834,24952,25362,40574/hello'
+    r = await client.get(
+        yarl.URL("/467%2C802%2C24834%2C24952%2C25362%2C40574/hello", encoded=True)
+    )
+    assert r.status == 200
+    await r.release()
+
+
+@pytest.mark.xfail(
+    raises=AssertionError,
+    reason="Regression in v3.7: https://github.com/aio-libs/aiohttp/issues/5621",
+)
+async def test_decoded_raw_match_regex(aiohttp_client) -> None:
+    app = web.Application()
+
+    async def handler(_):
+        return web.Response()
+
+    app.router.add_get("/467%2C802%2C24834%2C24952%2C25362%2C40574/hello", handler)
```

Comment:
Why doesn't this one have a regex? If the reproducer doesn't depend on it, why don't we just remove it from the first test too? Also, the URL is quite long and since the test doesn't seem to be influenced by its length, let's just make it 2-character, one char would be ascii and the other one would be something urlencoded — this will reduce the cognitive load for the readers.

Reply:
it doesn't need a regex, this is ensuring you can't have a match to an escaped path

Reply:
I have a feeling that the first test doesn't need that either. Just register a decoded route, right?

Reply:
correct, I've written it to match the actual bug in our production system.  I think it's worth keeping both ways to ensure both the regex works and the non regex, resulting in more code coverage

Reply:
ok cool, I wasn't sure if you guys like separate tests or combo tests

Reply:
Okay, then, please incorporate the earlier parametrization suggestions and keep both cases as params.

Reply:
done